src/os/bluestore/BlueRocksEnv.cc

```diff
@@ -499,6 +499,26 @@ rocksdb::Status BlueRocksEnv::LinkFile(
   ceph_abort();
 }
 
+rocksdb::Status BlueRocksEnv::AreFilesSame(
+  const std::string& first,
+  const std::string& second, bool* res)
+{
+  std::string dir, file;
+  split(first, &dir, &file);
+  int r;
+  r = fs->stat(dir, file, nullptr, nullptr);
+  if (r < 0) {
+    return err_to_status(r);
+  }
+  split(second, &dir, &file);
+  r = fs->stat(dir, file, nullptr, nullptr);
+  if (r < 0) {
+    return err_to_status(r);
+  }
+  *res = (first == second);
```

Comment:
Hi, Kefu,   it seems that `AreFilesSame` is used to judge whether first's ino is same with second's. see [posix_env's impl] (https://github.com/facebook/rocksdb/blob/master/env/env_posix.cc#L608). bluefs doesn't support [LinkFile](https://github.com/ceph/ceph/blob/master/src/os/bluestore/BlueRocksEnv.cc#L495-L500) now. so I prefer to return a IsNotSupported status here.

see https://github.com/facebook/rocksdb/blob/master/db/repair.cc#L258-L264, Rocksdb will do this check too.
```
    Status status = env_->AreFilesSame(db_options_.wal_dir, dbname_, &same);
    if (status.IsNotSupported()) {
      same = db_options_.wal_dir == dbname_;
      status = Status::OK();
    } else if (!status.ok()) {
      return status;
    }
```

Reply:
@Liuchang0812 i did read the related code. and i still lean towards the approach to to stat before comparing. if any of the file does not exist, we should error out.

Reply:
`GetChildren` will check whether file exist too. see https://github.com/facebook/rocksdb/blob/master/db/repair.cc#L271
https://github.com/ceph/ceph/blob/master/src/os/bluestore/BlueRocksEnv.cc#L417

Reply:
yeah, but i am trying to implement `BlueRocksEnv::AreFilesSame()` instead of working on a minimal patch.